cmd/pdcli/alerts/cmd.go

```diff
@@ -32,71 +36,110 @@ import (
 )
 
 type Alert struct {
-	IncidentID string
-	Name       string
-	ClusterID  string
-	Severity   string
-	Status     string
+	IncidentID  string
+	AlertID     string
+	ClusterID   string
+	ClusterName string
+	Name        string
+	Console     string
+	Labels      string
+	LastCheckIn string
+	Severity    string
+	Status      string
+	Sop         string
+	Token       string
+	Tags        string
 }
 
-var args struct {
-	high       bool
-	low        bool
-	assignment string
-	columns    string
+var options struct {
+	high        bool
+	low         bool
+	assignment  string
+	columns     string
+	interactive bool
+	incidentID  bool
 }
 
 var Cmd = &cobra.Command{
 	Use:   "alerts",
 	Short: "This command will list all the open high alerts assigned to self.",
+	Args:  cobra.MaximumNArgs(1),
 	RunE:  alertsHandler,
 }
 
 func init() {
 
 	// Urgency
-	Cmd.Flags().BoolVar(&args.low, "low", false, "View all low alerts")
-	Cmd.Flags().BoolVar(&args.high, "high", true, "View all high alerts")
+	Cmd.Flags().BoolVar(&options.low, "low", false, "View all low alerts")
+	Cmd.Flags().BoolVar(&options.high, "high", true, "View all high alerts")
 
 	// Incident Assignment
 	Cmd.Flags().StringVar(
-		&args.assignment,
+		&options.assignment,
 		"assigned-to",
 		"self",
 		"Filter alerts based on user or team",
 	)
 
 	// Columns displayed
 	Cmd.Flags().StringVar(
-		&args.columns,
+		&options.columns,
 		"columns",
-		"incident.id,name,cluster.id,status,severity",
+		"incident.id,cluster.name,alert,cluster.id,status,severity",
 		"Specify which columns to display separated by commas without any space in between",
 	)
+
+	// Interactive mode
+	Cmd.Flags().BoolVarP(
+		&options.interactive,
+		"interactive",
+		"i",
+		false,
+		"interactive alerts",
+	)
+
 }
 
 // alertsHandler is the main alerts command handler.
 func alertsHandler(cmd *cobra.Command, args []string) error {
 	var incidentAlerts []Alert
 	var alerts []Alert
+	var incidentID string
 
+	// Create a new pagerduty client
 	client, err := pdcli.NewConnection().Build()
 
 	if err != nil {
 		return err
 	}
 
-	// Get incident IDs
-	incidentIDs, err := GetIncidents(client)
+	// Check for incident ID argument
+	if len(args) > 0 {
+		incidentID = strings.TrimSpace(args[0])
+
+		// Validate the incident ID
+		match, _ := regexp.MatchString(constants.IncidentIdRegex, incidentID)
+
+		if !match {
```

Comment:
inplace of `!match` you could use the err returned from regexp.MatchString

Reply:
nitpick, but just as convension, err is more reliable and conventional

Reply:
`match` is validating the regex, so I would suggest using `match` as it is.
Tried using `err` and didn't get the expected output for the wrong incident ID.
```
./pdcli alerts P0X97E12
status code: 404
```